File: rust/src/models/term_rate.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::billing_period_enum::BillingPeriodEnum;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct TermRate {
    pub price: rust_decimal::Decimal,

    pub term: BillingPeriodEnum,
}

impl TermRate {
    pub fn new(price: rust_decimal::Decimal, term: BillingPeriodEnum) -> Self {
        Self { price, term }
    }
}

File: typescript/src/models/termRate.ts
// this file is @generated
import { type BillingPeriodEnum, BillingPeriodEnumSerializer } from "./billingPeriodEnum";

export interface TermRate {
  price: string;

  term: BillingPeriodEnum;
}

export const TermRateSerializer = {
  _fromJsonObject(object: any): TermRate {
    return {
      price: object["price"],
      term: BillingPeriodEnumSerializer._fromJsonObject(object["term"]),
    };
  },

  _toJsonObject(self: TermRate): any {
    return {
      price: self.price,
      term: BillingPeriodEnumSerializer._toJsonObject(self.term),
    };
  },
};
